### DISCUSSION TITLE
move column to workspace that doesn't exist

### DISCUSSION THREAD

**Discussion by Fuann-Kinoko** [YaLTeR/niri#210] [General]
Hello, I am deeply impressed by niri! This can be said to be the most impressive tiling window manager I have ever tried.

However, during the testing, I found some situations that did not quite align with my expectations.

assume there are `n` workspaces currently:

For the command `focus-workspace` `x`, if `x` refers to a workspace that has not been created yet, it indeed switches to an empty workspace. But it seems that once a window is created in this workspace, regardless of the value of `x`, the actual index of the current workspace will always be `n+1`.

Similarly, for the command `move-column-to-workspace` `x`, after moving to a new workspace, the index of the current workspace will also bee `n+1` instead of the specified `x`.

I understand that dynamic workspaces is a feature of niri, but there is still some inconsistency between the commands and their actual outcomes. 

Perhaps it would be worth considering renaming these commands, allowing workspace numbers to be discontinuous, or simply adding a comment in the configuration to clarify this behavior?

**Comment by YaLTeR** [comment]
> Hello, I am deeply impressed by niri! This can be said to be the most impressive tiling window manager I have ever tried.

Thanks! I'm glad to hear that you find it interesting.

> I found some situations that did not quite align with my expectations.

So, yeah. Niri has dynamic workspaces, and allowing skipped indices and whatnot sounds like it won't really work. (There's the named workspaces idea though: https://github.com/YaLTeR/niri/issues/201.)

The actions to focus and move by workspace index work "on top" of the dynamic workspaces, i.e. they don't do anything special there. When adding them I thought it would be better to operate on the last workspace if the index is too big, instead of just doing nothing. I'm not sure there's an alternative here when working with dynamic workspaces.

I agree it would make sense to document this in the default config though.

**Reply by YaLTeR** [reply]
Is this a clear explanation? https://github.com/YaLTeR/niri/commit/85be5f746cd0104eb83c73b228348a1125155a94

**Reply by Fuann-Kinoko** [reply]
sure!
